test/unit/test-main.js

```diff
@@ -0,0 +1,136 @@
+var sinon = require('sinon');
+var proxyquire = require('proxyquire');
+var request = require('supertest');
+
+// mock out express app to allow verification in tests
+var mockExpress = {
+  use: function() {},
+  listen: function() {},
+  static: function() {}
+};
+var mockExpressApp = function() {
+  return mockExpress;
+};
+var mockStaticHandler = function static(){};
+mockExpressApp.static = function() {
+  return mockStaticHandler;
+};
+
+// mock out fh-mbaas-api to allow verification in tests
+var mockMbaasApi;
+var mockSys;
+var mockFhMiddleware;
+var mockCloud;
+var mockError;
+var mockSysHandler = function sys(){};
+var mockMbaasHandler = function mbaas(){};
+var mockFhMiddlewareHandler = function fhmiddleware(){};
+var mockCloudHandler = function cloud(){};
+var mockErrorHandler = function errorHandler(){};
+var createMockMbaasApi = function() {
+  return {
+    mbaasExpress: sinon.spy(function() {
```

Comment:
For the supertest, proxyquire needs to globally mock the fh-mbaas-api with the '@global': true, flag

https://github.com/thlorenz/proxyquire#globally-override-require